### ISSUE TITLE
PHP Compile Error at Imagine\Imagick\Imagick::thumbnailImage

### ISSUE THREAD

**Issue by alexgit2k** [craftcms/happy-lager#34]
### Description

Calling homepage in browser says:
```
yii\base\ErrorException: Declaration of Imagine\Imagick\Imagick::thumbnailImage($columns, $rows, $bestfit = false, $fill = false, $filter = Imagick::FILTER_TRIANGLE, $keepImageProfiles = false, $keepExifData = false) must be compatible with Imagick::thumbnailImage(?int $columns, ?int $rows, bool $bestfit = false, bool $fill = false, bool $legacy = false): bool in craftcms/application/vendor/pixelandtonic/imagine/src/Imagick/Imagick.php:88
Stack trace:
#0 [internal function]: yii\base\ErrorHandler->handleFatalError()
#1 {main}
Copy StacktraceSearch StackoverflowSearch GoogleError
PHP Compile Error – yii\base\ErrorException
Declaration of Imagine\Imagick\Imagick::thumbnailImage($columns, $rows, $bestfit = false, $fill = false, $filter = Imagick::FILTER_TRIANGLE, $keepImageProfiles = false, $keepExifData = false) must be compatible with Imagick::thumbnailImage(?int $columns, ?int $rows, bool $bestfit = false, bool $fill = false, bool $legacy = false): bool
1. in craftcms/application/vendor/pixelandtonic/imagine/src/Imagick/Imagick.phpat line 88
79808182838485868788899091929394959697     * @param	bool	$bestfit   	    	Treat $columns and $rows as a bounding box in which to fit the image.
     * @param	bool	$fill		       	Fill in the bounding box with the background colour.
     * @param	integer	$filter			    The resampling filter to use. Refer to the list of filter constants at <http://php.net/manual/en/imagick.constants.php>.
     * @param   boolean $keepImageProfiles  Whether to keep ICP and ICM image profiles. Defaults to false.
     * @param   boolean $keepExifData       Whether to keep EXIF data. Defaults to false.
     *
     * @return	bool	Indicates whether the operation was performed successfully.
     */
 
    public function thumbnailImage($columns, $rows, $bestfit = false, $fill = false, $filter = \Imagick::FILTER_TRIANGLE, $keepImageProfiles = false, $keepExifData = false)
    {
 
        // sample factor; defined in original ImageMagick thumbnailImage function
        // the scale to which the image should be resized using the `sample` function
        $SampleFactor = 5;
 
        // filter whitelist
        $filters = array(
            \Imagick::FILTER_POINT,
2. yii\base\ErrorHandler::handleFatalError()
Yii Framework
2021-08-06, 14:44:39

nginx/1.20.1

Yii Framework/2.0.42.1
```

### Steps to reproduce

1. Setup
```
git clone https://github.com/craftcms/demo.git application
cd application
composer install
cp .env.example .env
sed -i 's/DB_DATABASE=.*/DB_DATABASE=craftcms/' .env
sed -i 's/DB_USER=.*/DB_USER=craftcms/' .env
sed -i 's/DB_PASSWORD=.*/DB_PASSWORD=craftcms/' .env
sed -i 's|SITE_URL=.*|SITE_URL=https://craftcms.xxx.yyy|' .env
php craft setup/security-key
mysql craftcms < happylager.sql
```
2. Call https://craftcms.xxx.yyy

### Additional info

- Craft version: latest version of happy-lager 
- PHP version: 8.0.9 as well as 7.4.22
- Database driver & version: MariaDB 10.6
- Plugins & versions:

**Comment by alexgit2k** [comment]
Probably related to https://github.com/craftcms/cms/issues/8095

However, pixelandtonic/imagine is installed in the latest available version 1.2.4.2

**Comment by alexgit2k** [comment]
It's working now after `composer update pixelandtonic/imagine`
Probably forgot to clear caches before.

However, pixelandtonic/imagine in composer.lock should be updated in this repo.
